addons/sale_coupon/models/coupon_program.py

```diff
@@ -42,7 +45,11 @@ def _check_promo_code(self, order, coupon_code):
             message = {'error': _('The promotional offer is already applied on this order')}
         elif not self.active:
             message = {'error': _('Promo code is invalid')}
-        elif self.rule_date_from and self.rule_date_from > order.date_order or self.rule_date_to and order.date_order > self.rule_date_to:
+        elif self.rule_date_from and self.rule_date_from > fields.Datetime.now():
+            user_tz = pytz.timezone(self.env.context.get('tz') or self.env.user.tz or 'UTC')
+            coupon_date = datetime.strftime(pytz.utc.localize(self.rule_date_from, DEFAULT_SERVER_DATETIME_FORMAT).astimezone(user_tz), "%d/%m/%Y %H:%M:%S")
```

Comment:
This seems overcomplicated & wrong no?
The format is hardcoded to `"%d/%m/%Y %H:%M:%S"` instead of the user one.

What about just 
```python
from babel.dates import format_datetime
tzinfo = self.env.context.get('tz') or self.env.user.tz or 'UTC'
locale = self.env.context.get('lang') or self.env.user.lang or 'en_US'
format_datetime(self.rule_date_from, format='short', tzinfo=tzinfo, locale=locale)
```
![image](https://user-images.githubusercontent.com/30048408/109987639-183fa780-7d07-11eb-8e0c-451f1b9f5783.png)